va/validation-authority.go

```diff
@@ -103,21 +104,30 @@ func (va ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdentif
 	}
 	hostName := identifier.Value
 
+	addrs, problem := va.getAddrs(identifier.Value)
+	if problem != nil {
+		challenge.Status = core.StatusInvalid
+		challenge.Error = problem
+		return challenge, challenge.Error
```

Comment:
Why are you returning challenge.Error when challenge already contains error on line 110.